### PR TITLE
DA-3418: Add common utilities for slack webhook notifications & fetching aws ssm parameter

### PR DESCRIPTION
**Pull request by ajinkyan83**
Ref: https://github.com/LF-Engineering/ledger-service/blob/master/util/parameters.go

Ref: sample util to send logs to slack: https://github.com/LF-Engineering/reimbursement-service/tree/master/slack

The logs will be sent to `insights-logs` slack channel

### CODE REVIEW

**Review by fayazg — APPROVED** [review]

File: aws/ssm/parameters.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	ssmsvc := ssm.New(sess, aws.NewConfig().WithRegion("us-west-2"))
+	param, err := ssmsvc.GetParameter(&ssm.GetParameterInput{
+		Name:           &key,
+		WithDecryption: &withDecryption,
+	})
+	if err != nil {
+		log.Fatal("Error getting SSM parameter:", err)
+		return "", err
+	}
+	return *param.Parameter.Value, nil
```
Comment (enkhalifapro) [inline aws/ssm/parameters.go:40]: unsafe access to pointer, You may check if it not nil first

**Review by code-sleuth — APPROVED** [review]

File: slack/slack.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	resty "gopkg.in/resty.v1"
+)
+
+var (
+	errorBadResponse = "API call gave an other-than-200 result"
+	errorEmptyText   = "Refusing to send an empty message, consider calling slack.SetText() first"
+)
+
+// API provides an interface to interact with the Slack Webhook API
+type API struct {
```
Comment (enkhalifapro) [inline slack/slack.go:16]: This name is very generic, you may add another descriptive name

File: slack/slack.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
+package slack
```
Comment (enkhalifapro) [inline slack/slack.go:1]: Why we have this in common-libraries?

File: slack/slack_test.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	api := getAPI(t)
+	err := api.Send()
+	assert.EqualError(t, err, errorEmptyText)
+}
+
+func TestSendNotEmpty(t *testing.T) {
+	api := getAPI(t)
+	api.SetText("TestSendNotEmpty Test Content")
+	err := api.Send()
+	assert.Nil(t, err, "unexpected err from api.Send()")
```
Comment (enkhalifapro) [inline slack/slack_test.go:32]: there's an assert check called noError which I think it's more suitable

File: aws/ssm/parameters.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+// from Parameter Store (Encrypted)
+// Variables in parameter store must all be encrypted
+func GetParameter(key string) (string, error) {
+	value := os.Getenv(key)
+	if value != "" {
+		return value, nil
+	}
+	withDecryption := true
+	sess, err := session.NewSessionWithOptions(session.Options{
+		Config:            aws.Config{Region: aws.String("us-west-2")},
```
Comment (lukaszgryglicki) [inline aws/ssm/parameters.go:23]: us-west-2 should go to some env variable IMHO

File: aws/ssm/parameters.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+		SharedConfigState: session.SharedConfigEnable,
+	})
+
+	if err != nil {
+		log.Fatal("Error getting a new session:", err)
+		return "", err
+	}
+	ssmsvc := ssm.New(sess, aws.NewConfig().WithRegion("us-west-2"))
```
Comment (lukaszgryglicki) [inline aws/ssm/parameters.go:31]: same here, say `AWS_REGION`.

File: slack/slack.go (hunk 2 above)
Comment (lukaszgryglicki) [reply slack/slack.go:16]: full type name (with `slack`) will be `slack.API` - not sure if still bad - have no preference.

File: aws/ssm/parameters.go (hunk 1 above)
Comment (ajinkyan83) [reply aws/ssm/parameters.go:45]: It is already checking for error returned in previous statement & accessing pointer if no error.

File: slack/slack.go (hunk 2 above)
Comment (ajinkyan83) [reply slack/slack.go:16]: Yes, I prefer to keep it is API considering the package name is `slack`.

File: slack/slack.go (hunk 3 above)
Comment (ajinkyan83) [reply slack/slack.go:1]: I thought of adding this here, considering other services or modules could re-use this.
They will just need to set their own custom webhook url.

File: slack/slack_test.go (hunk 4 above)
Comment (ajinkyan83) [reply slack/slack_test.go]: done.

File: aws/ssm/parameters.go (hunk 5 above)
Comment (ajinkyan83) [reply aws/ssm/parameters.go]: done.

File: aws/ssm/parameters.go (hunk 6 above)
Comment (ajinkyan83) [reply aws/ssm/parameters.go]: done

File: slack/slack.go (hunk 2 above)
Comment (enkhalifapro) [reply slack/slack.go:16]: You may commit go standard convention for services which is Doer, so I prefer something like `Provider`

File: aws/ssm/parameters.go (hunk 1 above)
Comment (enkhalifapro) [reply aws/ssm/parameters.go:45]: If you are sure it's fine

File: aws/ssm/parameters.go [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+		log.Fatal("Error getting a new session:", err)
+		return "", err
+	}
+	ssmsvc := ssm.New(sess, aws.NewConfig().WithRegion(region))
+	param, err := ssmsvc.GetParameter(&ssm.GetParameterInput{
+		Name:           &key,
+		WithDecryption: &withDecryption,
+	})
+	if err != nil {
+		log.Fatal("Error getting SSM parameter:", err)
```
Comment (enkhalifapro) [inline aws/ssm/parameters.go:42]: this Fatal will make process exit, is this what you want?

File: slack/slack.go (hunk 3 above)
Comment (fayazg) [reply slack/slack.go:1]: Will be same for all

File: slack/slack.go (hunk 2 above)
Comment (ajinkyan83) [reply slack/slack.go]: done!

File: aws/ssm/parameters.go (hunk 7 above)
Comment (ajinkyan83) [reply aws/ssm/parameters.go]: updated to warning

**Review by lukaszgryglicki — APPROVED** [review]
/lgtm - one very minor comment - I would put `us-west-2` in some env variable, like`AWS_REGION`.

**Review by enkhalifapro — APPROVED** [review]

**Pull request merged**